### PR TITLE
refactor(@clayui/management-toolbar): breaks ManagementToolbar into small blocks and remove high level

### LINKED ISSUE #2452: Verify clay-management-toolbar for 3.0.0 release
- [ ] Are you able to easily implement a demo of this component? Are there any “gotchas” when implementing? Provide a link to codesandbox for a demo.
- [ ] Does the component accept additional HTML attributes? For example, adding an aria attribute to a button.
- [ ] Does component accept a ref? Is the ref value what you would expect?
- [ ] Does this component allow for different translations?
- [ ] If this is a form element, does it accept a `name` attribute?
- [ ] Were you able to find documentation for this component? Was it clear?

Additional Questions:
- Is there anything about the public API that you would change?
- Is the naming of the available props clear?

### PR DESCRIPTION
**Pull request by matuzalemsteles**
Fixes #2452 #2292

I ended up following the idea I had about this component in the beginning but I am still maintaining its high-level component... we can think if it makes sense to maintain if we can't just go with the low-level and see how it will work.

The design for the low-level components:
```jsx
<ClayManagementToolbar>
	<ClayManagementToolbar.ItemList>
		<ClayManagementToolbar.Item>
			// DropDown or button, any
		</ClayManagementToolbar.Item>
		<ClayManagementToolbar.Item expand />
	</ClayManagementToolbar.ItemList>

	<ClayManagementToolbar.Search>
	</ClayManagementToolbar.Search>
</ClayManagementToolbar>

<ClayResultsBar>
	<ClayResultsBar.Item expand />
</ClayResultsBar>
```

### CODE REVIEW

File: packages/clay-management-toolbar/src/Search.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import classNames from 'classnames';
+import React from 'react';
+
+interface IProps extends React.FormHTMLAttributes<HTMLFormElement> {
+	onlySearch?: boolean;
+
+	showMobile?: boolean;
```
Comment (bryceosterhaus) [inline packages/clay-management-toolbar/src/Search.tsx]: Why is this exactly for mobile if the class name is just `show`? Maybe adding some docs to the interfaces might help

**Comment by bryceosterhaus** [comment]
@matuzalemsteles I was just re-looking at `ActiveState` and `DefaultState` and I think we should probably break those down further into smaller low-level components. They seem pretty large and fairly strict as to what they render. We will need to make sure they are low-level enough to provide all sorts of management bar arrangements, such as various elements or groupings of elements to not exist. For example, see the variations at https://liferay.design/lexicon/core-components/toolbars/management-bar/#layout-variations

**Comment by matuzalemsteles** [comment]
> @matuzalemsteles I was just re-looking at ActiveState and DefaultState and I think we should probably break those down further into smaller low-level components. They seem pretty large and fairly strict as to what they render. We will need to make sure they are low-level enough to provide all sorts of management bar arrangements, such as various elements or groupings of elements to not exist. For example, see the variations at https://liferay.design/lexicon/core-components/toolbars/management-bar/#layout-variations

Hmm what are you thinking? do you have any idea of ​​the API design for them, I don't know how we could break `ActiveState` and `DefaultState` to a low-level, I think we can make both low level but separate.

File: packages/clay-management-toolbar/src/ManagementToolbar.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import classNames from 'classnames';
+import Item from './Item';
+import ItemList from './ItemList';
+import React from 'react';
+import Search from './Search';
+
+interface IProps extends React.HTMLAttributes<HTMLElement> {
```
[the remark this answers is not in the document]
Comment (matuzalemsteles) [reply packages/clay-management-toolbar/src/ManagementToolbar.tsx]: I thought the same at first but it seems that there is no `HTMLNavElement` interface.

File: packages/clay-management-toolbar/src/ManagementToolbarWithElements.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+			{showResultsBar && (
+				<ClayResultsBar>
+					<ClayResultsBar.Item>
+						<span className="component-text text-truncate-inline">
+							<span className="text-truncate">
+								{sub(labels.totalItems, [totalItems])}
+								{searchValue && (
+									<>
+										{' "'}
```
[the remark this answers is not in the document]
Comment (matuzalemsteles) [reply packages/clay-management-toolbar/src/ManagementToolbarWithElements.tsx]: Yeah, just to create the space between `results for "Red"` for example.

File: packages/clay-management-toolbar/src/Search.tsx (hunk 1 above)
Comment (matuzalemsteles) [reply packages/clay-management-toolbar/src/Search.tsx]: Ideally it was meant to be clear, but I see it wasn't 😅... In the mobile view the search will disappear, so you have to add the show class to show the search in the mobile view. I will take care of adding docs in these interfaces.

**Comment by bryceosterhaus** [comment]
I think I would imagine a low-level API being used similarly to what you had posted. See a rough example below.

```jsx
<ClayManagementToolbar active={active}>
	{!active && (
		<>
			<ClayManagementToolbar.Group>
				<ClayManagementToolbar.SelectAll />
				<ClayManagementToolbar.Dropdown />
				<ClayManagementToolbar.Icon />
			</ClayManagementToolbar.Group>

			<ClayManagementToolbar.Search />

			<ClayManagementToolbar.Group>
				<ClayManagementToolbar.Icon />
				<ClayManagementToolbar.Icon />
				<ClayManagementToolbar.Dropdown />
			</ClayManagementToolbar.Group>
		</>
	)}

	{active && (
		<>
			<ClayManagementToolbar.Group>
				<ClayManagementToolbar.SelectAll />
				<ClayManagementToolbar.Text />
				<ClayManagementToolbar.Button />
				<ClayManagementToolbar.Button />
			</ClayManagementToolbar.Group>

			<ClayManagementToolbar.Group>
				<ClayManagementToolbar.Icon />
				<ClayManagementToolbar.Icon />
				<ClayManagementToolbar.Icon />
				<ClayManagementToolbar.Dropdown />
			</ClayManagementToolbar.Group>
		</>
	)}
</ClayManagementToolbar>
```

I think this is roughly what you have, and maybe we just need to remove the high-level portion. I think the management bar tends to vary from place to place too, so it might make sense to just keep it primarily low-level. Thinking we need to aim specifically for low-level to accommodate for all the variations seen at https://liferay.design/lexicon/core-components/toolbars/management-bar/#layout-variations.

**Comment by matuzalemsteles** [comment]
Hmm, I'm not sure if this would be the best. The concern is, this way we are just creating a wrapper on top of the `Icon`, `DropDown`, `Button` components so that we get the different variations and this increases the coupling of the components. I think it's a different way of just doing the same as low-level.

I think we can discuss if the ideal would be to send this high-level. I think it has use within the Portal, is a very common guy in most portlets, so defining via API how this high-level will behave among these scenarios I would be well.

**Comment by bryceosterhaus** [comment]
@matuzalemsteles yeah we wouldn't necessarily have to wrap `Icon`, `DropDown`, and `Button`, that would just be for the case if they need a specific class or wrapper around it. Such as, a ul for the grouping and li around each item.

As for high-level, maybe we can have a team using this component to first try the low-level and then we can evaluate the best way to move forward from there, that way we can get feedback without having to commit to a high-level release quite yet.

**Comment by matuzalemsteles** [comment]
> @matuzalemsteles yeah we wouldn't necessarily have to wrap Icon, DropDown, and Button, that would just be for the case if they need a specific class or wrapper around it. Such as, a ul for the grouping and li around each item.

Oh ok, it's clearer 🙂

> As for high-level, maybe we can have a team using this component to first try the low-level and then we can evaluate the best way to move forward from there, that way we can get feedback without having to commit to a high-level release quite yet.

Yeah, np. I can remove the high-level implementation of the management toolbar and we can bring it in at any time based on the feedbacks we collect. Can we go ahead with the low level and release it today?

**Comment by matuzalemsteles** [comment]
@bryceosterhaus updated, Let me know if you see anything else.

**Pull request merged**